add test_bun add test_burger add test_database.py add test_ingridient

Zorkiy87 opened the pull request.

Zorkiy87 committed:
add test_bun
add test_burger
add test_database.py
add test_ingridient

DolgoarshinnykhLG commented on tests/conftest.py:
```diff
@@ -0,0 +1,39 @@
+from praktikum.database import Database
+
+
+@pytest.fixture()
+def burger():
+    return Burger()
```
Нужно исправить: фикстуры не занимаются выполнением примитивной логики, они выполняют сложную логику предусловий\постусловий и вычислений. Эти объекты можно сразу создавать в тестах

Zorkiy87 replied:
Исправлено.

DolgoarshinnykhLG commented on tests/conftest.py:
```diff
@@ -0,0 +1,39 @@
+    other_ingredient_mock.get_price.return_value = 42.0
+    return other_ingredient_mock
+
+@pytest.fixture()
+def database():
+    return Database()
```
Нужно исправить: фикстуры не занимаются выполнением примитивной логики, они выполняют сложную логику предусловий\постусловий и вычислений. Эти объекты можно сразу создавать в тестах

Zorkiy87 replied:
Исправлено.

DolgoarshinnykhLG commented on tests/test_bun.py:
```diff
@@ -0,0 +1,25 @@
+
+class TestBun:
+    @pytest.mark.parametrize("name, price", [
+        ("Белая булка", 123.03),
+        ("Бородинский хлеб", 789.25),
+        ("Чиабата", 250.00),
+        ("Блины", 125.25)
```
Можно лучше: тестовые данные и ожидаемые ответы лучше не хранить в теле/модулях теста. Лучше вынести их в модуль data, так их будет легче поддерживать

DolgoarshinnykhLG commented on tests/test_burger.py:
```diff
@@ -0,0 +1,37 @@
+        burger.add_ingredient(ingredient_mock)
+        burger.add_ingredient(other_ingredient_mock)
+        receipt = burger.get_receipt()
+        assert "(==== hot bun ====)" and "= sause test sause =" and "= filling test filling =" and "Price: 174" in receipt
```
Нужно исправить: проверка через in в данном кейсе не будет надёжной, необходимо проверить полное соответствие

Zorkiy87 replied:
Исправлено. Тест переписаан. Убрана проверка через in.

DolgoarshinnykhLG commented on tests/test_ingridient.py:
```diff
@@ -0,0 +1,13 @@
+    ing = Ingredient(ingredient_type,name,price)
+    assert ing.get_type() == ingredient_type
+    assert ing.get_name() == name
+    assert ing.get_price() == price
```
Нужно исправить: стоит разделить эти тесты, юнит-тесты должны быть максимально атомарными

Zorkiy87 replied:
Исправлено. Тесты разиты на 3 атомарных теста.

Zorkiy87 committed:
1) Убраны фикстуры burger и database. Переписаны тесты на которые они влияли.
2) Изменён тест test_get_receipt, убрана проверка через in
3) Разделены тесты ингредиентов, сейчас отдельные: test_ingredient_init_type,  test_ingredient_init_name, test_ingredient_init_price

DolgoarshinnykhLG commented on tests/test_ingridient.py:
```diff
@@ -0,0 +1,28 @@
+    (INGREDIENT_TYPE_FILLING, 'big naggets', 250.20),
+    (INGREDIENT_TYPE_SAUCE, 'garlic sauce', 95.00)
+])
+def test_ingredient_init_type(ingredient_type,name, price):
```
Стоило их так же объединить в тестовый класс, как в соседних модулях

Zorkiy87 committed:
1) Тесты из test_ingredient объединены в тестовый класс TestIngredient
